OSDOCS-12237-nwt:nwt bug batch 4.18 (#88110)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

[id="ocp-release-note-networking-bug-fixes_{context}"]
==== Networking

> * The SR-IOV Network Operator supports Intel NetSec Accelerator Cards and Marvell Octeon 10 DPUs. (link:https://issues.redhat.com/browse/OCPBUGS-43451[*OCPBUGS-43451*])

Reviewer: This is labeled as an Enhancement. Should it be moved?

> * Previously, enabling encapsulated security payload (ESP) offload hardware when using IPSec on Open vSwitch attached interfaces would break connectivity in your cluster. To resolve this issue, {product-title} by default disables ESP hardware offload on Open vSwitch attached interfaces. This fixes the issue. (link:https://issues.redhat.com/browse/OCPBUGS-42987[*OCPBUGS-42987*])

Reviewer: ```suggestion
* Previously, enabling encapsulated security payload (ESP) hardware offload when using IPSec on Open vSwitch attached interfaces would break connectivity in your cluster. To resolve this issue, {product-title} by default disables ESP hardware offload on Open vSwitch attached interfaces. This fixes the issue. (link:https://issues.redhat.com/browse/OCPBUGS-42987[*OCPBUGS-42987*])
```

* Previously, if you set custom annotations in a custom resource (CR), the SR-IOV Operator would override all the default annotations in the `SriovNetwork` CR. With this release, when you define custom annotations in a CR, the SR-IOV Operator does not override the default annotations. (link:https://issues.redhat.com/browse/OCPBUGS-41352[*OCPBUGS-41352*])

* Previously, bonds that were configured in `active-backup` mode would have IPsec Encapsulating Security Payload (ESP) offload active even if underlying links did not support ESP offload. This caused IPsec associations to fail. With this release, ESP offload is disabled for bonds so that IPsec associations pass. (link:https://issues.redhat.com/browse/OCPBUGS-39438[*OCPBUGS-39438*])

> * When you start the limited live migration method and an issue exists with network overlap, the Cluster Network Operator (CNO) can now expose network overlap metrics for the issue. This is possible because the `openshift_network_operator_live_migration_blocked` metric now includes the new `NetworkOverlap` label. (link:https://issues.redhat.com/browse/OCPBUGS-39096[*OCPBUGS-39096*])

Reviewer: Should this go under New Features and Enhancements?

* Previously, the Machine Config Operator (MCO)'s vSphere `resolve-prepender` script used `systemd` directives that were incompatible with old bootimage versions used in {product-title} 4. With this release, nodes can scale using newer bootimage versions {product-version} 4.13 and above, through manual intervention, or by upgrading to a release that includes this fix. (link:https://issues.redhat.com/browse/OCPBUGS-38012[*OCPBUGS-38012*])

* Previously, the Ingress Controller status incorrectly displayed as `Degraded=False` because of a migration time issue with the `CanaryRepetitiveFailures` condition. With this release, the Ingress Controller status is correctly marked as `Degraded=True` for the appropriate length of time that the `CanaryRepetitiveFailures` condition exists. (link:https://issues.redhat.com/browse/OCPBUGS-37491[*OCPBUGS-37491*])

> Previously, when a pod was running on a node on which egress IPv6 is assigned, the pod was not able to communicate with the Kubernetes service in a dual stack cluster. This resulted in the traffic with the IP family, that the egressIP is not applicable to, being dropped. With this release, only the source network address translation (SNAT) for the IP family that the egress IPs applied to is deleted, eliminating the risk of traffic being dropped. (link:https://issues.redhat.com/browse/OCPBUGS-37193[*OCPBUGS-37193*])

Reviewer: ```suggestion
* Previously, when a pod was running on a node on which egress IPv6 is assigned, the pod was not able to communicate with the Kubernetes service in a dual stack cluster. This resulted in the traffic with the IP family, that the egressIP is not applicable to, being dropped. With this release, only the source network address translation (SNAT) for the IP family that the egress IPs applied to is deleted, eliminating the risk of traffic being dropped. (link:https://issues.redhat.com/browse/OCPBUGS-37193[*OCPBUGS-37193*])
```

* Previously, the Single-Root I/O Virtualization (SR-IOV) Operator did not expire the acquired lease during the Operator's shutdown operation. This impacted a new instance of the Operator, because the new instance had to wait for the lease to expire before the new instance was operational. With this release, an update to the Operator shutdown logic ensures that the Operator expires the lease when the Operator is shutting down. (link:https://issues.redhat.com/browse/OCPBUGS-23795[*OCPBUGS-23795*])

> The current PTP grandmaster clock (T-GM) implementation has a single National Marine Electronics Association (NMEA) sentence generator sourced from the GNSS without a backup NMEA sentence generator. If NMEA sentences are lost before reaching the e810 NIC, the T-GM cannot synchronize the devices in the network synchronization chain and the PTP Operator reports an error. A proposed fix is to report a `FREERUN` event when the NMEA string is lost. Until this limitation is addressed, T-GM does not support PTP clock holdover state. (link:https://issues.redhat.com/browse/OCPBUGS-19838[*OCPBUGS-19838*])

Reviewer: ```suggestion
* The current PTP grandmaster clock (T-GM) implementation has a single National Marine Electronics Association (NMEA) sentence generator sourced from the GNSS without a backup NMEA sentence generator. If NMEA sentences are lost before reaching the e810 NIC, the T-GM cannot synchronize the devices in the network synchronization chain and the PTP Operator reports an error. A proposed fix is to report a `FREERUN` event when the NMEA string is lost. Until this limitation is addressed, T-GM does not support PTP clock holdover state. (link:https://issues.redhat.com/browse/OCPBUGS-19838[*OCPBUGS-19838*])
```
Should probably go in Known Issues.

* Previously, when the live migration to OVN-Kubernetes was employed on clusters using the subnet 100.88.0.0/16 for either the `clusterNetwork` or `serviceNetwork` IP address pools, the `ovnkube-node` pod would crash after migration. With this update, the issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-43740[*OCPBUGS-43740*])

* Previously, a change in OVN-Kubernetes that standardized the `appProtocol` value `h2c` to `kubernetes.io/h2c` was not recognized by OpenShift router. Consequently, specifying `appProtocol: kubernetes.io/h2c` on a service did not cause OpenShift router to use clear-text HTTP/2 to connect to the service endpoints. With this release, OpenShift router was changed to handle `appProtocol: kubernetes.io/h2c` the same way as it handles `appProtocol: h2c` resolving the issue. (link:https://issues.redhat.com/browse/OCPBUGS-42972[*OCPBUGS-42972*])

> * Previously, instructions that guided the user after changing the `LoadBalancer` parameter from `External` to `Internal` were missing for PowerVS, Alibaba, and OpenStack platform. This caused the Ingress Controller to be put in a permanent `Progressing` state. With this release the message `The IngressController scope was changed from Internal to External" is followed by `To effectuate this change, you must delete the service` resolving the permanent `Progressing` state. (link:https://issues.redhat.com/browse/OCPBUGS-39151[*OCPBUGS-39151*])

Reviewer: ```suggestion
* Previously, instructions that guided the user after changing the `LoadBalancer` parameter from `External` to `Internal` were missing for PowerVS, Alibaba, and OpenStack platform. This caused the Ingress Controller to be put in a permanent `Progressing` state. With this release the message `The IngressController scope was changed from Internal to External` is followed by `To effectuate this change, you must delete the service` resolving the permanent `Progressing` state. (link:https://issues.redhat.com/browse/OCPBUGS-39151[*OCPBUGS-39151*])
```

> * Previously, instructions that guided the user after changing the `LoadBalancer` parameter from `External` to `Internal` were missing for PowerVS, Alibaba, and OpenStack platform. This caused the Ingress Controller to be put in a permanent `Progressing` state. With this release the message `The IngressController scope was changed from Internal to External" is followed by `To effectuate this change, you must delete the service` resolving the permanent `Progressing` state. (link:https://issues.redhat.com/browse/OCPBUGS-39151[*OCPBUGS-39151*])

Reviewer: ```suggestion
* Previously, instructions that guided the user after changing the `LoadBalancer` parameter from `External` to `Internal` were missing for :ibm-power-server-title:, :alibaba:, and :rh-openstack-first:. This caused the Ingress Controller to be put in a permanent `Progressing` state. With this release the message `The IngressController scope was changed from Internal to External" is followed by `To effectuate this change, you must delete the service` resolving the permanent `Progressing` state. (link:https://issues.redhat.com/browse/OCPBUGS-39151[*OCPBUGS-39151*])
```

* A regression in the behaviour of `libreswan` caused some nodes with IPsec enabled to lose communication with pods on other nodes in the same cluster. To resolve this issue, consider disabling IPsec for your cluster. (link:https://issues.redhat.com/browse/OCPBUGS-43713[*OCPBUGS-43713*])

> * There is a known issue with RHEL 8 worker nodes that  use `cgroupv1` Linux Control Groups (cgroup). The following is an example of the error message displayed for impacted nodes: `UDN are not supported on the node ip-10-0-51-120.us-east-2.compute.internal as it uses cgroup v1.` As a workaround, users should migrate worker nodes from `cgroupv1` to `cgroupv2`. (link:https://issues.redhat.com/browse/OCPBUGS-49933[*OCPBUGS-49933*])

Reviewer: ```suggestion
* There is a known issue with :op-system-base: 8 worker nodes that  use `cgroupv1` Linux Control Groups (cgroup). The following is an example of the error message displayed for impacted nodes: `UDN are not supported on the node ip-10-0-51-120.us-east-2.compute.internal as it uses cgroup v1.` As a workaround, users should migrate worker nodes from `cgroupv1` to `cgroupv2`. (link:https://issues.redhat.com/browse/OCPBUGS-49933[*OCPBUGS-49933*])
```

* A regression in the behaviour of `libreswan` caused some nodes with IPsec enabled to lose communication with pods on other nodes in the same cluster. To resolve this issue, consider disabling IPsec for your cluster. (link:https://issues.redhat.com/browse/OCPBUGS-43713[*OCPBUGS-43713*])

> * There is a known issue with :op-base-system: 8 worker nodes that  use `cgroupv1` Linux Control Groups (cgroup). The following is an example of the error message displayed for impacted nodes: `UDN are not supported on the node ip-10-0-51-120.us-east-2.compute.internal as it uses cgroup v1.` As a workaround, users should migrate worker nodes from `cgroupv1` to `cgroupv2`. (link:https://issues.redhat.com/browse/OCPBUGS-49933[*OCPBUGS-49933*])

Reviewer 2: probably {op-base-system} 8. Not rendering as-is

* The current PTP grandmaster clock (T-GM) implementation has a single National Marine Electronics Association (NMEA) sentence generator sourced from the GNSS without a backup NMEA sentence generator. If NMEA sentences are lost before reaching the e810 NIC, the T-GM cannot synchronize the devices in the network synchronization chain and the PTP Operator reports an error. A proposed fix is to report a `FREERUN` event when the NMEA string is lost. Until this limitation is addressed, T-GM does not support PTP clock holdover state. (link:https://issues.redhat.com/browse/OCPBUGS-19838[*OCPBUGS-19838*])

> * There is a known issue with a Layer 2 network topology on clusters running on Google Cloud Platform (GCP). At this time, the egress IP addresses being used in the Layer 2 network that is created by a `UserDefinedNetwork` (UDN) resource are using the wrong source IP address. Consequentially, UDN is not supported on Layer 2 on GCP. Currently, there is no fix for this issue. (link:https://issues.redhat.com/browse/OCPBUGS-48301*OCPBUGS-48301*])

Reviewer 2: Missing bracket is causing this to rendering incorrectly

```suggestion
* There is a known issue with a Layer 2 network topology on clusters running on Google Cloud Platform (GCP). At this time, the egress IP addresses being used in the Layer 2 network that is created by a `UserDefinedNetwork` (UDN) resource are using the wrong source IP address. Consequentially, UDN is not supported on Layer 2 on GCP. Currently, there is no fix for this issue. (link:https://issues.redhat.com/browse/OCPBUGS-48301[*OCPBUGS-48301*])
```

> * There is a known issue when a user-defined network (UDN) is enabled, OVN-Kubernetes deletes any routing table ID equal or higher to 1000 that it does not manage. Consequently, any Virtual Routing and Forwarding (VRF) instance created outside OVN-Kubernetes is deleted. This issue impacts users who have created user-defined VRFs with a table ID greater than or equal to 1000. As a workaround, users must change their VRFs to a table ID lower than 1000 or greater than 1 million as these are reserved for {product-title}. (link:https://issues.redhat.com/browse/OCPBUGS-50855[*OCPBUGS-50855*])

Reviewer 3: thats true

> * There is a known issue when a user-defined network (UDN) is enabled, OVN-Kubernetes deletes any routing table ID equal or higher to 1000 that it does not manage. Consequently, any Virtual Routing and Forwarding (VRF) instance created outside OVN-Kubernetes is deleted. This issue impacts users who have created user-defined VRFs with a table ID greater than or equal to 1000. As a workaround, users must change their VRFs to a table ID lower than 1000 or greater than 1 million as these are reserved for {product-title}. (link:https://issues.redhat.com/browse/OCPBUGS-50855[*OCPBUGS-50855*])

Reviewer 4: one small nit: the doc states "There is a known issue when a user-defined network (UDN) is enabled", but this is the default and not opt-in right?

> * There is a known issue when a user-defined network (UDN) is enabled, OVN-Kubernetes deletes any routing table ID equal or higher to 1000 that it does not manage. Consequently, any Virtual Routing and Forwarding (VRF) instance created outside OVN-Kubernetes is deleted. This issue impacts users who have created user-defined VRFs with a table ID greater than or equal to 1000. As a workaround, users must change their VRFs to a table ID lower than 1000 or greater than 1 million as these are reserved for {product-title}. (link:https://issues.redhat.com/browse/OCPBUGS-50855[*OCPBUGS-50855*])

Reviewer 3: ya, its the default.

> * There is a known issue when a user-defined network (UDN) is enabled, OVN-Kubernetes deletes any routing table ID equal or higher to 1000 that it does not manage. Consequently, any Virtual Routing and Forwarding (VRF) instance created outside OVN-Kubernetes is deleted. This issue impacts users who have created user-defined VRFs with a table ID greater than or equal to 1000. As a workaround, users must change their VRFs to a table ID lower than 1000 or greater than 1 million as these are reserved for {product-title}. (link:https://issues.redhat.com/browse/OCPBUGS-50855[*OCPBUGS-50855*])

Author: @martinkennelly and @fedepaol is this language more accurate? I'd like to merge this PR by EOD.

> * There is a known issue when a user-defined network (UDN) is enabled, OVN-Kubernetes deletes any routing table ID equal or higher to 1000 that it does not manage. Consequently, any Virtual Routing and Forwarding (VRF) instance created outside OVN-Kubernetes will be deleted. This issue impacts users who have created user-defined VRFs with a table ID greater than or equal to 1000. As a workaround, users must change their VRFs to a table ID lower than 1000 or greater than 1 million as these are reserved for {product-title}. (link:https://issues.redhat.com/browse/OCPBUGS-50855*OCPBUGS-50855*])

Reviewer 2: ```suggestion
* There is a known issue when a user-defined network (UDN) is enabled, OVN-Kubernetes deletes any routing table ID equal or higher to 1000 that it does not manage. Consequently, any Virtual Routing and Forwarding (VRF) instance created outside OVN-Kubernetes is deleted. This issue impacts users who have created user-defined VRFs with a table ID greater than or equal to 1000. As a workaround, users must change their VRFs to a table ID lower than 1000 or greater than 1 million as these are reserved for {product-title}. (link:https://issues.redhat.com/browse/OCPBUGS-50855[*OCPBUGS-50855*])
```
